src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -81,29 +84,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String user = userService.getCurrentUser().getEmail();
     String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    UrlValidator defaultValidator = new UrlValidator();
-
+    //Checking text against set regex. If there is a match, variable url is set to equal it
     String regex = "(https?://\\S+\\.(png|jpg|jpeg|gif|tiff|bmp)\\S*)";
 
-    System.out.println("Text is:" + userText);
-
     Pattern pattern = Pattern.compile(regex);
     Matcher matcher = pattern.matcher(userText);
     String url = "";
     while (matcher.find()) {
       url = matcher.group();
     }
 
+    //Validating the URL
+    UrlValidator defaultValidator = new UrlValidator();
+
     if (defaultValidator.isValid(url)) {
       String replacement = "<img src=\"$1\" />";
       String textWithImagesReplaced = userText.replaceAll(regex, replacement);
       Message message = new Message(user, textWithImagesReplaced);
-      datastore.storeMessage(message);
     } else {
         Message message = new Message(user, userText);
-        datastore.storeMessage(message);
     }
 
+    datastore.storeMessage(message);
```

Comment:
wait uhh does this work? it seems like the message variable is out of scope here? My intellj interpreter flags this message variable as undeclared in this scope.